File: Source/Core/AuthyToken.cpp
#include "AuthyToken.hpp"

AuthyToken::AuthyToken()
{
    // set Authy TOTP defaults
    _type = Authy;
    _name = "Authy";

    _digits = 7U;
    _period = 10U;
    _counter = 0U; // counter is not used in TOTP
    _algorithm = SHA1;
}

AuthyToken::AuthyToken(const Label &label)
    : AuthyToken()
{
    _label = label;
}

File: Android/app/src/main/java/com/magiruuvelvet/otpgen/Tokens/AuthyToken.kt
package com.magiruuvelvet.otpgen.Tokens

class AuthyToken : TOTPToken()
{
    init
    {
        _digits = 7;
        _period = 10;
        _algorithm = Companion.ShaAlgorithm.SHA1;
    }

    override val _type: Companion.Type = Companion.Type.Authy;

    override fun setCounter(counter: Int) {}
    override fun setPeriod(period: Int) {}
    override fun setDigits(digits: Int) {}
    override fun setAlgorithm(algo: Companion.ShaAlgorithm) {}
}
